devito/arch/compiler.py

```diff
@@ -243,6 +245,20 @@ def version(self):
 
         return version
 
+    @property
+    def _complex_ctype(self):
```

Comment:
no, we definitely don't want code-generation-related machinery in our Compiler classes.

The right thing to do is, instead, single-dispatching the Compiler class within our own compilation pass, which is responsible for the lowering of complex

Reply:
as such, I don't think we need to add a custom `name` to Compiler?

Reply:
I don't quite agree here, the complex type are defined by the actual compiler and their standard, i.e gnu has `_Complex` and cpp has `std::complex`, adding complicated dispatch is overkill for something that is standardized at the language level

Reply:
I disagree, singledispatch would achieve the same exact objective by dispatching based on the type. 

Instead, what you've done here violates a crucial principle of the OO paradigm, that is, classes should have a well defined purpose. These classes are for jit-compiling a given string. They're not supposed to provide compiler-specific code generation (C- or C++ specific) information

> adding complicated dispatch

I don't think it's complicated at all. An Iet_pass receives the `compiler` and all you have to do is creating a series of functions based on single dispatch doing the same exact thing it's being done here

Reply:
>g. They're not supposed to provide compiler-specific code generation (C- or C++ specific) information

But that's not what it is, this defines the standard associated with the compiler which is c99->_Complex, c++11->std:complex

adding a pass that move the standard out of the compiler doesn't really make sense.

Reply:
I didn't say (I hope) to add a pass. A single-dispatch function to retrieve some sort of type-specific information doesn't have to be a compiler pass.

But obviously our compiler pass would use it to get the code it needs